lib/rubocop/config.rb

```diff
@@ -509,12 +509,14 @@ def validate_section_presence(name)
     def validate_parameter_names(valid_cop_names)
       valid_cop_names.each do |name|
         validate_section_presence(name)
+        default_configuration = ConfigLoader.default_configuration[name]
         self[name].each_key do |param|
           next if COMMON_PARAMS.include?(param) ||
-                  ConfigLoader.default_configuration[name].key?(param)
+                  default_configuration.key?(param)
 
-          warn Rainbow("Warning: unrecognized parameter #{name}:#{param} " \
-                       "found in #{smart_loaded_path}").yellow
+          warn Rainbow("Warning: #{name} does not support #{param} parameter." \
```

Comment:
Nice! Perhaps we can go one step further by:

1. Filtering out parameters that are for internal use.
2. Formatting the parameters into a list.

So the output would look like:

```
Warning: Metrics/LineLength does not support Min parameter.

Supported parameters are: 

  - Enabled
  - Max
  - AllowHeredoc
  - AllowURI
  - URISchemes
  - IgnoreCopDirectives
  - IgnoredPatterns
```

WDYT?

Reply:
@Drenmi yeah, I think it will be more readable this way.
I have made your suggested update, kindly check that